### PR TITLE
Add overview panel

### LINKED ISSUE #450: Overview panel
It would be helpful to have an overview panel like this one in the top left of the image that shows the overall image and a little box covering where the current field of view is. I believe @manthey already has something for this for histomicstk that he showed me recently.

image

### PR DESCRIPTION
**Pull request by bruyeret**
Close #450

### CODE REVIEW

**Review by arjunrajlab — APPROVED** [review]
Tested and it looks great to me!

**Comment by arjunrajlab** [comment]
Actually, I'm seeing a slight problem with this file:

https://www.dropbox.com/scl/fi/ola4b1iphhgenl98zxkca/20230807_133502_470_LVE016-HKC-1dpi-HSV-1-WT-ICP4_GFP-MX1_Cy3-IFNb_A594-IRF3_Cy5-DDX58_Cy7_Stitched.nd2?rlkey=2unlgixyv2pn6x7zi3odgcqgy&dl=0

If you load that file, the overview doesn't quite match the image. The overview cuts off the bottom and right.

**Review by arjunrajlab — CHANGES_REQUESTED** [review]
Actually, I'm seeing a slight problem with this file:

https://www.dropbox.com/scl/fi/ola4b1iphhgenl98zxkca/20230807_133502_470_LVE016-HKC-1dpi-HSV-1-WT-ICP4_GFP-MX1_Cy3-IFNb_A594-IRF3_Cy5-DDX58_Cy7_Stitched.nd2?rlkey=2unlgixyv2pn6x7zi3odgcqgy&dl=0

If you load that file, the overview doesn't quite match the image. The overview cuts off the bottom and right.

**Review by arjunrajlab — APPROVED** [review]
Just tested and it is working great, thanks! Fine to merge for me.

**Comment by bruyeret** [comment]
It should be fixed now @arjunrajlab 
Thank you @manthey !

image

File: src/components/ImageOverview.vue [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+  downState:
+    | (IPanInfo & {
+        state: any; // geo.actionState
+        mouse: any; // geo.mouseState
+        distanceToOutline: number;
+      })
+    | null = null;
+
+  readonly directionItems: [boolean, boolean][] = [
+    [true, true],
```
Comment (luciemac) [inline src/components/ImageOverview.vue:59]: Can you add some comments?

**Comment by luciemac** [comment]
then, LGTM

**Comment by bruyeret** [comment]
I added a comment and made the code clearer, do you see another place where the code would need comments?

**Comment by luciemac** [comment]
nope looks great now ! Thanks

**Pull request merged**